File: app/src/main/java/com/karanchuk/roman/testtranslate/ui/translator/TranslatorContract.java
package com.karanchuk.roman.testtranslate.ui.translator;

import android.content.Context;

import com.karanchuk.roman.testtranslate.ui.base.BaseView;
import com.karanchuk.roman.testtranslate.data.database.model.DictDefinition;
import com.karanchuk.roman.testtranslate.data.database.model.TranslatedItem;

import java.util.List;

/**
 * Created by roman on 11.12.17.
 */

public interface TranslatorContract {

    interface Presenter {

        void attachView(Context context);
        void detachView();

        boolean requestTranslatorAPI();
        void requestDictionaryAPI();

        void vocalizeSourceText();
        void vocalizeTargetText();

        void recognizeSourceText();
        void resetRecognizer();

        void saveToSharedPreferences();
        void handleDictionaryResponse(DictDefinition dictDefinition);
        void clearContainerSuccess();

        List<TranslatedItem> getHistoryTranslatedItems();
    }

    interface View extends BaseView<Presenter> {

        void setHintOnInput();

        TranslatedItem createPredictedTranslatedItem();
        void getTranslatedItemFromCache(TranslatedItem maybeExistedItem);

        String getTextButtonSrcLang();
        String getTextButtonTrgLang();

        void setTextButtonSrcLang(final String text);
        void setTextButtonTrgLang(final String text);

        boolean isEmptyTranslatedResultView();
        String getTextTranslatedResultView();

        void setTextCustomEditText(final String text);
        boolean isEmptyCustomEditText();
        void clearCustomEditText();

        boolean isRecognizingSourceText();

        void requestRecordAudioPermissions();
        boolean isRecordAudioGranted();

        void showLoadingDictionary();
        void hideLoadingDictionary();

        void showRetry();
        void hideRetry();

        void showSuccess();
        void hideSuccess();
        void showError();

        void showActiveInput();
        void hideActiveInput();

        void showKeyboard();
        void hideKeyboard();

        void showClear();
        void hideClear();

        void showLoadingTargetVoice();
        void hideLoadingTargetVoice();

        void showIconTargetVoice();
        void hideIconTargetVoice();

        void showLoadingSourceVoice();
        void hideLoadingSourceVoice();

        void showIconSourceVoice();
        void hideIconSourceVoice();

        void activateVoiceRecognizer();
        void deactivateVoiceRecognizer();

        void stopAnimationMicroWaves();
        void showAnimationMicroWaves();
    }
}

File: app/src/main/java/com/romankaranchuk/translator/ui/translator/TranslatorContract.kt
package com.romankaranchuk.translator.ui.translator

import com.romankaranchuk.translator.data.database.model.TranslatedItem

interface TranslatorContract {

    interface View {
        fun setHintOnInput()
        fun createPredictedTranslatedItem(): TranslatedItem
        fun setTextButtonSrcLang(text: String)
        fun getTextButtonSrcLang(): String
        fun setTextButtonTrgLang(text: String)
        fun getTextButtonTrgLang(): String
        fun isEmptyTranslatedResultView(): Boolean
        fun getTextTranslatedResultView(): String
        fun setTextCustomEditText(text: String)
        fun isEmptyCustomEditText(): Boolean
        fun clearCustomEditText()
        fun isRecognizingSourceText(): Boolean
        fun requestRecordAudioPermissions()
        fun isRecordAudioGranted(): Boolean
        fun showLoadingDictionary()
        fun hideLoadingDictionary()
        fun showRetry()
        fun hideRetry()
        fun showSuccess()
        fun hideSuccess()
        fun showError()
        fun showActiveInput()
        fun hideActiveInput()
        fun showKeyboard()
        fun hideKeyboard()
        fun showClear()
        fun hideClear()
        fun showLoadingTargetVoice()
        fun hideLoadingTargetVoice()
        fun showIconTargetVoice()
        fun hideIconTargetVoice()
        fun showLoadingSourceVoice()
        fun hideLoadingSourceVoice()
        fun showIconSourceVoice()
        fun hideIconSourceVoice()
        fun activateVoiceRecognizer()
        fun deactivateVoiceRecognizer()
        fun stopAnimationMicroWaves()
        fun showAnimationMicroWaves()
    }
}